python/python/lance/dataset.py

```diff
@@ -538,6 +540,33 @@ def restore(self):
         """
         self._ds.restore()
 
+    def cleanup_old_versions(self, before: datetime = None) -> CleanupStats:
+        """
+        Cleans up old versions of the dataset.
+
+        Some dataset changes, such as a delete, remove data from the dataset but
+        do not physically remove the deleted rows from storage.  The old data is
```

Comment:
I worry about delete being the primary example here, since in many cases deletion will add a bitmap, but cleanup won't remove anything. Maybe overwrite is a better example?

Reply:
I've changed to overwrite